test/integration/backward_compatible/parallel/serialization/compact/CompactUtil.js

```diff
@@ -343,6 +349,60 @@ if (TestUtil.isClientVersionAtLeast('5.1.0')) {
         ], field: Fields.ARRAY_OF_GENERIC_RECORD},
     };
 
+    SampleObject1 = class SampleObject1 {
+        constructor(type, id) {
+            this.type = type; // string
```

Comment:
IMO, calling this field "type" makes it confusing. It can be a `name` or something like this. The real type comes from the `Schema.typeName` or the class of the object.